### DISCUSSION TITLE
Relational data in JSON import file

### DISCUSSION THREAD

**Discussion by tomniemi** [directus/directus#15307] [Archive]
Hi,

I am able to import items one collection at a time from JSON file. 

But how to import child items to child collection at a same time as importing parent items? E.g if I have 'test_parent' parent collection and 'test_child' child collection, how to import 'test_child' items inside 'test_parent' item? 

I tried something like this, but it only adds parent items ('test_parent_id' is set to have a relationship to test_parent.id):

```
[    
    {
        "id": 1,
        "test": "TEST 1",
        "test_child": 
        [
            {
                "id": 2,
                "test_parent_id": 1,
                "test": "CHILDTEST 1"
            }
        ]
    }
]

```

**Comment by br41nslug** [comment]
It depends a bit on what type of relation it is for the example i'll use an M2M:
```json
[

	{
		"related_field": [
			{ // create a new item by values
				"other_collection": {
					"test_a": "xxxxxx",
					"test_b": "yyyyy",
					"test_c": "zzzzz",
				}
			},
			{ // link an existing item in the other collection
				"other_collection": {
					"id": 123,
				}
			},
		]
	},
]
```
I don't think the reverse field `"test_parent_id": 1,` is needed in most cases.

**Reply by tomniemi** [reply]
@br41nslug thank you for your answer! I am using one-to-many relationship (one 'test_parent' can have multiple 'test_child' items).

If I inspect Directus log, it shows "insert into 'test_parent'..." row, but no "insert into 'test_child'" row.

**Reply by tomniemi** [reply]
@br41nslug I cannot find the documentation about how to specify relations in import file. So in case of one-to-many relationship, I tried to refer to child collection using it's name (in this case 'test_child'), but apparently it is not the correct syntax:

`[    
    {        
        "test1": "TEST 1dd",        
        "test_child": 
        [
            {                
                "childtest": "CHILDTEST 1dd"
            }
        ]
    }    
]`

Ouput I excepted was that one 'test_parent' row was added and also one 'test_child' row and 'test_child.test_parent_id' field value contained the primary key value of 'test_parent'.

**Reply by br41nslug** [reply]
I doubt there are specific import docs on that but it should definitely follow the logic of the API https://docs.directus.io/reference/introduction.html#relational-data
> Output I excepted was that one 'test_parent' row was added and also one 'test_child' row and 'test_child.test_parent_id' field value contained the primary key value of 'test_parent'.

And what was the result you got?

**Reply by br41nslug** [reply]
What does the format look like if you export it with the related fields you want? when ignoring IDs that should be importable 
![image](https://user-images.githubusercontent.com/9389634/187209342-3eba8125-03de-4375-86f4-73538f543316.png)

**Reply by tomniemi** [reply]
> 
> > Output I excepted was that one 'test_parent' row was added and also one 'test_child' row and 'test_child.test_parent_id' field value contained the primary key value of 'test_parent'.
> 
> And what was the result you got?

On 'test_parent' row was added. And Directus log also showed that only sql insert statement for 'test_parent' collection was executed.

**Reply by tomniemi** [reply]
@br41nslug Thank you for your help! I managed to import relational data by looking at the export results. It turns out that I have named Alias field in parent table differently from the child collection name and if I used that Alias name to reference child collection everything works.

**Comment by rijkvanzanten** [comment]
(@erondpowell relational export/import would be good to highlight in the docs)
